tenant/repository.go

```diff
@@ -51,6 +63,15 @@ func (s DBService) GetTenant(tenantID uuid.UUID) (*Tenant, error) {
 	return &t, nil
 }
 
+func (s DBService) NamespaceExists(nsName string) bool {
+	var ns Namespace
+	err := s.db.Table(Namespace{}.TableName()).Where("name = ?", nsName).Find(&ns).Error
+	if err != nil {
+		return false
+	}
+	return true
```

Comment:
Check if the error is NotFoundError. If yes then return false. Otherwise return the error.